client/src/js/services/grid/GridColumns.js

```diff
@@ -85,7 +87,10 @@ function GridColumnService(uiGridConstants, AppCache, Modal, util) {
   function Columns(gridOptions, cacheKey) {
 
     // set up local storage for selected grid columns
-    this.cache = AppCache(cacheKey + serviceKey);
+    if(cacheKey){
```

Comment:
Out of curiosity, why is a cache optional?  It seems to me that it would be cleaner if you emitted a warning and defaulted to a value.